### PR TITLE
Symmetry-corrected RMSD calculations

### PR DESCRIPTION
**Pull request by RMeli**
This PR is an attempt to add a `SymmRMSD` class for symmetry-corrected RMSD calculations, powered by [spyrmsd](https://github.com/RMeli/spyrmsd). For the time being, I created a separate class from the main`RMSD` class because I believe the scope of symmetry-corrected RMSD calculations is much more narrow (and because there are some open issues with the current `RMSD` class, such as https://github.com/MDAnalysis/mdanalysis/issues/2891).

Tests are currently missing, see https://github.com/MDAnalysis/MDAnalysisData/issues/45.

Example usage:
```python
import MDAnalysis as mda
from MDAnalysis.analysis import align, rms

from matplotlib import pyplot as plt

mobile = mda.Universe("npt_prod1.tpr", "npt_prod1.xtc")
ref = mda.Universe("npt_prod1.tpr", "npt_prod1.xtc", frame=0)

R = rms.RMSD(mobile, ref, select='name CA', groupselections=["resname LIG"])
R.run()

# Align trajectory to reference
# This is done internally by rms.RMSD, but it is not accessible
# See https://github.com/MDAnalysis/mdanalysis/issues/2891
_ = align.AlignTraj(mobile, ref, select='name CA',in_memory=True).run()

sR = rms.SymmRMSD(mobile, ref, select="resname LIG")
sR.run()

plt.figure()
plt.plot(R.results.rmsd[:,0], R.results.rmsd[:,-1], label="RMSD")
plt.plot(sR.results.rmsd[:,0], sR.results.rmsd[:,-1], label="SymmRMSD")
plt.legend()
plt.savefig("rmsd.png")
```

This is still very much WIP, but early comments/suggestions/criticisms are very welcome.

TODOs:
* [ ] Allow to compute weighted RMSD
* [ ] Documentation

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?

### CODE REVIEW

File: package/MDAnalysis/analysis/rms.py [hunk 1]
```diff
@@ -158,6 +158,8 @@
 import logging
 import warnings
 
+import spyrmsd.rmsd
```
Comment (IAlibay) [inline package/MDAnalysis/analysis/rms.py]: Let's not make this a core dependency, guard this import? (see MSD)

File: package/MDAnalysis/analysis/rms.py [hunk 2]
```diff
@@ -711,6 +713,107 @@ def rmsd(self):
         return self.results.rmsd
 
 
+class SymmRMSD(AnalysisBase):
```
Comment (IAlibay) [inline package/MDAnalysis/analysis/rms.py:756]: Don't forget the relevant duecredit stuff

**Review by IAlibay — COMMENTED** [review]
Just a couple of pre review comments

File: package/MDAnalysis/analysis/rms.py [hunk 3]
```diff
@@ -711,6 +713,107 @@ def rmsd(self):
...
+    """
+
+    def __init__(self, atomgroup, reference=None, select='all', ref_frame=0, **kwargs):
+        super().__init__(atomgroup.universe.trajectory, **kwargs)
+
+        self.atomgroup = atomgroup
+        self.reference = reference if reference is not None else self.atomgroup
+        self.ref_frame = ref_frame
+
+        select = process_selection(select)
```
Comment (richardjgowers) [inline package/MDAnalysis/analysis/rms.py]: Can we move away from the pattern of passing in selection strings? IMHO more preferable is AtomGroups (which you can make with selection strings if that works for you)

File: package/MDAnalysis/analysis/rms.py [hunk 4]
```diff
@@ -711,6 +713,107 @@ def rmsd(self):
...
+        """
+        n_atoms = len(atoms)
+
+        # Allocate adjacency matrix
+        A = np.zeros((n_atoms, n_atoms), dtype=int)
+
+        # TODO: Catch NoDataError in case of missing bonds?
+        # FIXME: Make this more efficient
+        # Loop over all bonds
+        for bond in atoms.bonds:
```
Comment (richardjgowers) [inline package/MDAnalysis/analysis/rms.py]: Rocco my eyes 😂 A[atoms.bonds.to_indices()] = 1 or something will work

Comment (richardjgowers) [reply package/MDAnalysis/analysis/rms.py]: Ah but you're remapping, so you'll have to have a remap array too...

Comment (RMeli) [reply package/MDAnalysis/analysis/rms.py]: lol, I need a pre-PR check that spots the `FIXME`s...

Comment (RMeli) [reply package/MDAnalysis/analysis/rms.py]: Crappy implementation:
```
27.1 ms ± 277 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```
Vectorized implementation:
```
125 µs ± 1.81 µs per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
```

**Comment by RMeli** [comment]
Closing in favor of a MDAKit (WIP): https://github.com/RMeli/spyrmsdkit

**Pull request closed without merging**